src/common.rs

```diff
@@ -268,10 +271,30 @@ fn scalar_apply<C: FromIterator<Option<I>>, I>(
             return exec_err!("unexpected json array type {:?}", other);
         }
     );
-
     to_array(c)
 }
 
+fn prepare_dict<T: ArrowDictionaryKeyType>(
+    json_array: &DictionaryArray<T>,
+    values: ArrayRef,
+    return_dict: bool,
+) -> DataFusionResult<ArrayRef> {
+    if return_dict {
+        if is_json_union(values.data_type()) {
+            // JSON union: post-process the array to set keys to null where the union member is null
```

Comment:
We could probably move this logic into `json_get` since it only applies there.

cc @davidhewitt?